### ISSUE TITLE
Missing readline support

### ISSUE THREAD

**Issue by adamjstewart** [adamjstewart/prolog#3]
It would be great if the interpreter had readline support, i.e. the arrow keys allow us to cycle through past input or move left and edit something that has been typed.

**Comment by ZaidQureshi** [comment]
Completed. Refer to Commit: 8d0495aa7bf3fb42345a9d56a4e1944cd743e1de

**Comment by adamjstewart** [comment]
It looks like readline support was removed in 5cb3bf909f9c4daeaedd9630de9999620be53e2e?

**Comment by ZaidQureshi** [comment]
Yeah, there was a bug, if i get time ill put it in again although it is very low on my priority list right now.
